test/integration/install_test.go

```diff
@@ -163,6 +163,86 @@ func TestRetrieveUidPreUpgrade(t *testing.T) {
 	}
 }
 
+func TestInstallCNIPlugin(t *testing.T) {
+	if !TestHelper.CNI() {
+		return
+	}
+
+	// install the CNI plugin in the cluster
+	var (
+		cmd  = "install-cni"
+		args = []string{
+			"--use-wait-flag",
+			"--cni-log-level=debug",
+		}
+	)
+
+	exec := append([]string{cmd}, args...)
+	out, stderr, err := TestHelper.LinkerdRun(exec...)
+	if err != nil {
+		testutil.AnnotatedFatalf(t, "'linkerd install-cni' command failed",
+			"'linkerd install-cni' command failed: \n%s\n%s", out, stderr)
+	}
+
+	out, err = TestHelper.KubectlApply(out, "")
+	if err != nil {
+		testutil.AnnotatedFatalf(t, "'kubectl apply' command failed",
+			"'kubectl apply' command failed\n%s", out)
+	}
+
+	// perform a linkerd check with --linkerd-cni-enabled
+	timeout := time.Minute
+	err = TestHelper.RetryFor(timeout, func() error {
+		out, stderr, err = TestHelper.LinkerdRun("check", "--pre", "--linkerd-cni-enabled", "--wait=0")
+		if err != nil {
+			return fmt.Errorf("'linkerd check' command failed\n%s\n%s\n%v", out, stderr, err)
+		}
+		return nil
+	})
+	if err != nil {
+		testutil.AnnotatedFatal(t, fmt.Sprintf("'linkerd check' command timed-out (%s)", timeout), err)
+	}
+}
+
+func TestCalicoInstall(t *testing.T) {
```

Comment:
Wouldn't it make more sense to install Calico before linkerd's CNI plugin? (thus first `TestCalicoInstall` and then `TestInstallCNIPlugin`)

Reply:
Yep, Correct.